### PR TITLE
Migrate to idiomatic error handling in support_package.go

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
This PR contains a couple of changes to the support package code:
- Update `support_package.go` to use idiomatic error handling [c5a7d54](https://github.com/mattermost/mattermost/pull/24382/commits/c5a7d54b07bd73c93d492d9fa44b4f7a114349a8)
- Add `bleve_post_indexing` jobs to support package [47b8818](https://github.com/mattermost/mattermost/pull/24382/commits/47b88184593dd8703877779a2b8c6cb255d3bbfa)
- Reorder code to match `model/job.go` for better readability [c78a72e](https://github.com/mattermost/mattermost/pull/24382/commits/c78a72ea6c65b884f848bbbda065f709fb17c94f)

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-54245

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: server/channels/app/support_packet.go [hunk 1]
```diff
@@ -78,21 +80,36 @@ func (a *App) generateSupportPacketYaml() (*model.FileData, string) {
...
+		rErr = multierror.Append(errors.Wrap(err, "error while getting message export jobs"))
+	}
+	dataRetentionJobs, err := a.Srv().Store().Job().GetAllByTypePage(model.JobTypeDataRetention, 0, 2)
+	if err != nil {
+		rErr = multierror.Append(errors.Wrap(err, "error while getting data retention jobs"))
+	}
+	complianceJobs, err := a.Srv().Store().Job().GetAllByTypePage("compliance", 0, 2)
+	if err != nil {
+		rErr = multierror.Append(errors.Wrap(err, "error while getting compliance jobs"))
+	}
```
Comment (hanzei) [inline server/channels/app/support_packet.go:108]: There is no `compliance` job in https://github.com/mattermost/mattermost/blob/e7c7ce2286a393734ec42ec7c208e6773d61d41d/server/public/model/job.go#L11-L38.

@coltoneshaw Which job type do you mean here? `message_export`?

Comment (coltoneshaw) [reply server/channels/app/support_packet.go]: Hmm. I don't recall, very well could have been an error. Message export exists already, see below. I was trying to get a different job type that I pulled from the jobs table of a customer. I'm vauge and if it doesn't exist and was incorrect, just delete it. 

```
message_export_jobs:
- id: uqfwwzcqatng5gffmse6rmhmyh
  type: message_export
  priority: 0
  createat: 1693487253804
  startat: 0
  lastactivityat: 0
  status: pending
  progress: 0
  data: {}
  ```

**Review by coltoneshaw — APPROVED** [review]
Looks good!

File: server/channels/app/support_packet.go [hunk 2]
```diff
@@ -103,140 +124,129 @@ func (a *App) generateSupportPacketYaml() (*model.FileData, string) {
...
+	return fileData, nil
 
-	return nil, warning
 }
 
-func (a *App) getMattermostLog() (*model.FileData, string) {
-	var warning string
+func (a *App) getNotificationsLog() (*model.FileData, error) {
+	if !*a.Config().NotificationLogSettings.EnableFile {
+		return nil, errors.New("Unable to retrieve notifications.log because LogSettings: EnableFile is false in config.json")
```
Comment (isacikgoz) [inline server/channels/app/support_packet.go:206]: It's trivial but config doesn't always needs to be in `config.json`. We also have a config in the DB.

File: server/channels/app/support_packet.go [hunk 3]
```diff
@@ -103,140 +124,129 @@ func (a *App) generateSupportPacketYaml() (*model.FileData, string) {
...
-			fileData := model.FileData{
-				Filename: "mattermost.log",
-				Body:     mattermostLogFileData,
-			}
-			return &fileData, ""
-		}
-		warning = fmt.Sprintf("os.ReadFile(mattermostLog) Error: %s", mattermostLogFileDataErr.Error())
+func (a *App) getMattermostLog() (*model.FileData, error) {
+	if !*a.Config().LogSettings.EnableFile {
+		return nil, errors.New("Unable to retrieve mattermost.log because LogSettings: EnableFile is false in config.json")
```
Comment (isacikgoz) [inline server/channels/app/support_packet.go:224]: Same as above.

File: server/channels/app/support_packet.go (hunk 1 above)
Comment (hanzei) [reply server/channels/app/support_packet.go]: Deleted in f100553d94

File: server/channels/app/support_packet.go (hunk 2 above)
Comment (hanzei) [reply server/channels/app/support_packet.go]: See [3d4a97f](https://github.com/mattermost/mattermost/pull/24382/commits/3d4a97f831143cf8414678768d4700c052cc799f)

File: server/channels/app/support_packet.go (hunk 3 above)
Comment (hanzei) [reply server/channels/app/support_packet.go]: See See [3d4a97f](https://github.com/mattermost/mattermost/pull/24382/commits/3d4a97f831143cf8414678768d4700c052cc799f)

**Review by isacikgoz — APPROVED** [review]
Awesome, thanks @hanzei

**Comment by hanzei** [comment]
Note to self: Check docs

**Comment by hanzei** [comment]
https://docs.mattermost.com/manage/generating-support-packet.html does not need an update

**Pull request merged**